log/lvl.go

```diff
@@ -0,0 +1,385 @@
+// current debug-level set. Furthermore a set of common messages exist
+// that are printed according to a chosen format.
+//
+// The log-level messages are:
+//	log.Lvl1("Important information")
+//	log.Lvl2("Less important information")
+//	log.Lvl3("Eventually flooding information")
+//	log.Lvl4("Definitively flooding information")
+//	log.Lvl5("I hope you never need this")
+// in your program, then according to the debug-level one or more levels of
+// output will be shown. To set the debug-level, use
+//	log.SetDebugVisible(3)
+// which will show all `Lvl1`, `Lvl2`, and `Lvl3`. If you want to turn
+// on just one output, you can use
+//	log.LLvl2("Less important information")
+// By adding a single 'L' to the method, it *always* gets printed.
+//
+// You can also add a 'f' to the name and use it like fmt.Printf:
+//	log.Lvlf1("Level: %d/%d", now, max)
+//
+// The common messages are:
+//	log.Print("Simple output")
+//	log.Info("For your information")
+//	log.Warn("Only a warning")
+//	log.Error("This is an error, but continues")
+//	log.Panic("Something really went bad - calls panic")
+//	log.Fatal("No way to continue - calls os.Exit")
+//
+// These messages are printed according to the value of 'Format':
+// - Format == FormatLvl - same as log.Lvl
+// - Format == FormatPython - with some nice python-style formatting
+// - Format == FormatNone - just as plain text
+//
+// The log-package also takes into account the following environment-variables:
+//	DEBUG_LVL // will act like SetDebugVisible
+//	DEBUG_TIME // if 'true' it will print the date and time
+//	DEBUG_COLOR // if 'false' it will not use colors
+// But for this the function ParseEnv() or AddFlags() has to be called.
+package log
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/daviddengcn/go-colortext"
+)
+
+// These are information-debugging levels that can be turned on or off.
+// Every logging greater than 'DebugVisible' will be discarded. So you can
+// Log at different levels and easily turn on or off the amount of logging
+// generated by adjusting the 'DebugVisible' variable.
+var debugVisible = 1
+
+// If showTime is true, it will print the time for each line of debug-output
+var showTime = false
+
+// If useColors is true, debug-output will be colored
+var useColors = true
+
+var debugMut sync.RWMutex
+
+// NamePadding - the padding of functions to make a nice debug-output - this is automatically updated
+// whenever there are longer functions and kept at that new maximum. If you prefer
+// to have a fixed output and don't remember oversized names, put a negative value
+// in here
+var NamePadding = 40
+
+// LinePadding of line-numbers for a nice debug-output - used in the same way as
+// NamePadding
+var LinePadding = 3
+
+// Testing variable can have multiple values
+// 0 - no testing
+// 1 - put all line-numbers to 0
+// 2 - like 1, but write to TestString instead of stdout
+var Testing = 0
```

Comment:
yes - before `lvl_test.go` was defined as `package lvl_test`, so I needed it to be exported.